nautobot_ssot/integrations/vsphere/diffsync/adapters/adapter_vsphere.py

```diff
@@ -0,0 +1,312 @@
+    return ip_address
+
+
+def get_disk_total(disks: List):
+    """Calculcate total disk capacity."""
+    total = 0
+    for disk in disks:
+        total += disk["value"]["capacity"]
+    return int(total / 1024.0**3)
+
+
+class VsphereDiffSync(Adapter):
+    """vSphere adapter for DiffSync."""
+
+    clustergroup = ClusterGroupModel
+    cluster = ClusterModel
+    virtual_machine = VirtualMachineModel
+    interface = VMInterfaceModel
+    ip_address = IPAddressModel
+    prefix = PrefixModel
+
+    top_level = ["prefix", "clustergroup", "virtual_machine"]
+
+    def __init__(self, *args, job=None, sync=None, client, config, cluster_filters, **kwargs):
+        """Initialize the vSphereDiffSync."""
+        super().__init__(*args, **kwargs)
+        self.job = job
+        self.sync = sync
+        self.client = client
+        self.config = config
+        self.cluster_filters = cluster_filters
+
+    def _add_diffsync_virtualmachine(self, virtual_machine, virtual_machine_details, cluster_name):
+        """Add virtualmachine to DiffSync and call load_vm_interfaces().
+
+        Args:
+            virtual_machine (dict): Virtual Machine information from vSphere.
+            virtual_machine_details (dict): Virtual Machine detail information from vSphere.
+            diffsync_cluster (ClusterModel): Current DiffSync Cluster object.
+            cluster_name (str): Name of the cluster to which the VM belongs.
+        """
+        diffsync_virtualmachine, _ = self.get_or_instantiate(
+            self.virtual_machine,
+            {"name": virtual_machine.get("name"), "cluster__name": cluster_name},
+            {
+                "vcpus": virtual_machine.get("cpu_count", 0),
+                "memory": virtual_machine.get("memory_size_MiB", 0),
+                "disk": (
+                    get_disk_total(virtual_machine_details["disks"]) if virtual_machine_details.get("disks") else None
+                ),
+                "status__name": self.config.default_vm_status_map[virtual_machine["power_state"]],
+            },
+        )
+        self.load_vm_interfaces(
+            vsphere_virtual_machine_details=virtual_machine_details,
+            vm_id=virtual_machine["vm"],
+            diffsync_virtualmachine=diffsync_virtualmachine,
+        )
+
+    def load_cluster_groups(self):
+        """Load Cluster Groups (DataCenters)."""
+        clustergroups = self.client.get_datacenters().json()["value"]
+        self.job.log_debug(message=f"Loading ClusterGroups {clustergroups}")
+        for clustergroup in clustergroups:
+            self.get_or_instantiate(self.clustergroup, {"name": clustergroup["name"]})
+        return clustergroups
+
+    def load_virtualmachines(self, cluster, diffsync_cluster):
+        """Load Virtual Machines."""
+        virtual_machines = self.client.get_vms_from_cluster(cluster["cluster"]).json()["value"]
+        self.job.log_debug(message=f"Loading VirtualMachines from Cluster {cluster}: {virtual_machines}")
```

Comment:
This goes for all instances of the log_debug or log_*.

Reply:
All my other non-debug logs should be correct. [Example](https://github.com/nautobot/nautobot-app-ssot/blob/a101761f9c9d8286d504e1136eb6e72a87595ca1/nautobot_ssot/integrations/vsphere/diffsync/adapters/adapter_vsphere.py#L310)